tiledb/array.py

```diff
@@ -243,3 +251,748 @@ def index_domain_subarray(array, dom, idx: tuple):
+    :param Ctx ctx: TileDB context
+    """
+
+    def __init__(
+        self, uri, mode="r", key=None, timestamp=None, attr=None, ctx=None, **kwargs
+    ):
+        if ctx is None:
+            ctx = default_ctx()
+
+        self.array = kwargs.get(
+            "preloaded_array", preload_array(uri, mode, key, timestamp, ctx)
+        )
+
+        # view on a single attribute
+        schema = self.array._schema()
+        if attr is not None and not schema._has_attribute(attr):
+            self.array._close()
+            raise KeyError(f"No attribute matching '{attr}'")
+        else:
+            self.view_attr = attr
+
+        self.ctx = ctx
+        self.key = key
+        self.uri = uri
+        self.domain_index = DomainIndexer(self)
+        self.pyquery = None
+        self.__buffers = None
+
+        self.last_fragment_info = dict()
+        self.meta = Metadata(self)
+
+    def __capsule__(self):
+        return self.array.__capsule__()
+
+    def __repr__(self):
+        if self.isopen:
+            array_type = "Sparse" if self.schema.sparse else "Dense"
+            return f"Array(type={array_type }, uri={self.uri!r}, mode={self.mode}, ndim={self.schema.ndim})"
+        else:
+            return f"Array(uri={self.uri!r}, mode=closed)"
+
+    @classmethod
+    def create(cls, uri, schema, key=None, overwrite=False, ctx=None):
+        """Creates a TileDB Array at the given URI
+
+        :param str uri: URI at which to create the new empty array.
+        :param ArraySchema schema: Schema for the array
+        :param str key: (default None) Encryption key to use for array
+        :param bool overwrite: (default False) Overwrite the array if it already exists
+        :param Ctx ctx: (default None) Optional TileDB Ctx used when creating the array,
+                        by default uses the ArraySchema's associated context
+                        (*not* necessarily ``tiledb.default_ctx``).
+
+        """
+        if ctx is not None and not isinstance(ctx, Ctx):
+            raise TypeError(
+                "tiledb.Array.create() expected tiledb.Ctx " "object to argument ctx"
+            )
+        if ctx is None:
+            ctx = schema.ctx
+
+        from .dense_array import DenseArrayImpl
+        from .sparse_array import SparseArrayImpl
+
+        if issubclass(cls, DenseArrayImpl) and schema.sparse:
+            raise ValueError(
+                "Array.create `schema` argument must be a dense schema for DenseArray and subclasses"
+            )
+        if issubclass(cls, SparseArrayImpl) and not schema.sparse:
+            raise ValueError(
+                "Array.create `schema` argument must be a sparse schema for SparseArray and subclasses"
+            )
+
+        if key is not None:
+            config = tiledb.Config(ctx.config())
+            config["sm.encryption_type"] = "AES_256_GCM"
+            config["sm.encryption_key"] = key
+            ctx = tiledb.Ctx(config)
```

Comment:
```suggestion
```

Reply:
This is necessary to prevent the following error:

`E       tiledb.cc.TileDBError: [TileDB::Array] Error: ArrayDirectory: GenericTileIO: Error reading generic tile; tile is encrypted with NO_ENCRYPTION but given key is for AES_256_GCM`

Reply:
we need to create a new `Ctx` object with the modified `config`.